src/components/multi-select/multi-select.test.jsx

```diff
@@ -0,0 +1,500 @@
+import React, { createRef } from 'react';
+import {
+  render,
+  screen,
+  waitFor,
+} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import MultiSelect from './multi-select.jsx';
+import Tag from '../tag/tag.jsx';
+import {
+  getAutocompleter,
+  getAvailableOption,
+  getRemoveButton,
+  getSelectedOption,
+  openOptions,
+  queryAvailableOption,
+  querySelectedOption,
+  queryRemoveButton,
+} from './test-queries.js';
+
+describe('<MultiSelect>', () => {
+  const requiredProps = {
+    options: ['Foo', 'Bar'],
+    id: 'test-id',
+    label: 'test label',
+    name: 'field-id',
+  };
+
+  it('has defaults', () => {
+    const ref = createRef();
+
+    render(<MultiSelect {...requiredProps} ref={ref} />);
+    expect(document.body).toMatchSnapshot();
+    expect(ref.current).toMatchSnapshot();
+  });
+
+  describe('classNames API', () => {
+    it('uses class name overrides provided', () => {
+      render(<MultiSelect
+        {...requiredProps}
+        classNames={{
+          container: 'unique-container',
+          formControlChildrenContainer: 'unique-form-control-children-container',
+          input: 'unique-input-class',
+          tagList: 'unique-tag-list-class',
+        }}
+      />);
+
+      expect(document.querySelector('.unique-container')).toBeInTheDocument();
+      expect(document.querySelector('.unique-form-control-children-container')).toBeInTheDocument();
+      expect(document.querySelector('.unique-input-class')).toBeInTheDocument();
+      expect(document.querySelector('.unique-tag-list-class')).toBeInTheDocument();
+    });
+  });
+
+  describe('id API', () => {
+    it('sets id of container and selected options', () => {
+      render(<MultiSelect {...requiredProps} id="unique-id" value={['Foo']} />);
+
+      expect(document.querySelector('#unique-id')).toBeInTheDocument();
```

Comment:
I think it's indeed sixes, but if we wanna go with a particular style, I can definitely change this.

Reply:
I'm more of a fan of `document`. 😅  I dislike the `container` queries in general because it can miss element that are rendered outside the component (with things like a React Portal). 

As for this particular test, I think this is a smell that our role is on the wrong item. I've been looking at the ARIA implementation of comboboxes for a few weeks now and I think the `role="combobox"` is _suppose_ to be on the container. 

However, that is the W3 ARIA 1.1 specifications. What we have now is W3 ARIA 1.0 specifications. So either is fine but I think if we implement 1.1 in the future then the test would be more readable: 

```suggestion
      render(<MultiSelect {...requiredProps} id="unique-id" value={['Foo']} />);

      expect(screen.getByRole('combobox')).toHaveAttribute('id', 'unique-id');
```